api/app/Notifications/PoolCandidateStatusChanged.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Notifications;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Notifications\Notification;
+
+class PoolCandidateStatusChanged extends Notification
+{
+    use Queueable;
+
+    /**
+     * Create a new notification instance.
+     */
+    public function __construct(
+        public string $old_status,
```

Comment:
If a user doesn't have a status to start (not sure the scenario is true but it is happening in tests) then this fails a type check. I think accepting `string | null` should work.

Reply:
Good catch!
Ref: [handle null statuses](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7233/commits/aeb0ef6398ac68089569169f021fedf61779a98d)